data_structure/linked_list.cpp

```diff
@@ -66,33 +80,36 @@ void search(int x) {
         }
 }
 
-void show() {
-        node *t = start;
+template  <class X>
+void linked_list<X>::show() {
+        node <X>*t = start;
         while (t != NULL) {
                 std::cout << t->val << "\t";
                 t = t->next;
         }
 }
-
-void reverse() {
-        node *first = start;
-        if (first != NULL) {
-                node *second = first->next;
-                while (second != NULL) {
-                        node *tem = second->next;
-                        second->next = first;
-                        first = second;
-                        second = tem;
-                }
-                start->next = NULL;
-                start = first;
-        } else {
-                std::cout << "\nEmpty list";
-        }
+template  <class X>
+void linked_list<X>::reverse() {
+    node <X>*first = start;
+    if (first != NULL) {
+            node<X> *second = first->next;
+            while (second != NULL) {
+                    node<X> *tem = second->next;
+                    second->next = first;
+                    first = second;
+                    second = tem;
+            }
+            start->next = NULL;
+            start = first;
+    } else {
+            std::cout << "\nEmpty list";
+    }
 }
 
+
 int main() {
         int choice, x;
+        linked_list<int> object;
         do {
                 std::cout << "\n1. Insert";
                 std::cout << "\n2. Delete";
```

Comment:
Add `endl;` after each statement.

Reply:
Adding "endl" would have no meaning if we have "\n" at the beginning of the statements. Running it on my console just prints them perfectly.

Reply:
Agreed. Bu I feel using `endl;`is much better than `"\n";`.
<a href="https://www.geeksforgeeks.org/endl-vs-n-in-cpp/#:~:text=endl%20vs%20%5Cn%20in%20C%2B%2B,-Although%20they%20both&text=cout%20%3C%3C%20endl%20%3A%20Inserts%20a,Only%20inserts%20a%20new%20line.&text=cout%20%3C%3C%20'%5Cn',flushing%20of%20stream%20is%20required.">**Refer**</a>
So whenever dynamic allocation is done, It's always better to use `endl;` instead of `"\n";`

Reply:
i personally prefer to use '\n' because it is fast as it does not flushes the stream, but using '\n' statements become a bit unclear in reading

Reply:
I don't have problem with "\n"
But  in this program use fflush(stdout) and fflush(stdin) to clear buffer sometimes wrong output may come due to unclear buffer. 
<a href="https://www.geeksforgeeks.org/clearing-the-input-buffer-in-cc/">Refer</a>

Reply:
yes you are right @mishraabhinn that in case of char arrays or strings '\n' creates a problem if we do not remove '\n' form the buffer or in other words it becomes necessary to flush the buffer but this all happens when we use C function `scanf()` or in C++ if we use something like `cin.getline()` but in this implementaion we are not using any of these functions we are taking input with `std::cin` only so i think it will not cause such buffer issues. well i don't have any problem in using `std::endl`.

@ananyasharma1442 please replace '\n' with `std::endl`